frontend/src/scenes/pipeline/hogfunctions/HogFunctionStatusIndicator.tsx

```diff
@@ -0,0 +1,117 @@
+                config. Updating your function will move it back to the "degraded" state for testing. If it performs
+                well, it will then be moved to the healthy.
+            </>
+        ),
+    },
+}
+
+const DEFAULT_DISPLAY: DisplayOptions = {
+    tagType: 'default',
+    display: 'Unknown',
+    description: (
+        <>The function status is unknown. The status will be derived once enough invocations have been performed.</>
+    ),
+}
+
+export function HogFunctionStatusIndicator(): JSX.Element | null {
+    const { hogFunction } = useValues(pipelineHogFunctionConfigurationLogic)
+
+    if (!hogFunction || !hogFunction.enabled) {
+        return null
+    }
+
+    const { tagType, display, description } = hogFunction.status?.state
+        ? displayMap[hogFunction.status.state]
+        : DEFAULT_DISPLAY
+
+    const averageRating = hogFunction.status?.ratings.length
+        ? hogFunction.status.ratings.reduce((acc, x) => acc + x.rating, 0) / hogFunction.status.ratings.length
+        : 0
+
+    return (
+        <LemonDropdown
+            overlay={
+                <>
+                    <div className="p-2 space-y-2">
+                        <h2 className="flex items-center m-0 gap-2">
+                            Function status - <LemonTag type={tagType}>{display}</LemonTag>
+                        </h2>
+
+                        <p>
+                            Your function has a rating of <b>{Math.round(averageRating * 100)}%</b>. A rating of 100%
+                            means the function is running perfectly, with 0% meaning it is failing every time.
+                        </p>
+
+                        <p>{description}</p>
+
+                        <h4>History</h4>
+                        <ul>
+                            <LemonTable
+                                columns={[
+                                    {
+                                        title: 'Timestamp',
+                                        key: 'timestamp',
+                                        render: (_, { timestamp }) => <TZLabel time={dayjs(timestamp)} />,
+                                    },
+                                    {
+                                        title: 'Status',
+                                        key: 'state',
+                                        render: (_, { state }) => {
+                                            const { tagType, display } = displayMap[state] || DEFAULT_DISPLAY
+                                            return <LemonTag type={tagType}>{display}</LemonTag>
+                                        },
+                                    },
+                                ]}
+                                dataSource={hogFunction.status?.states ?? []}
+                            />
+                        </ul>
+                    </div>
+                </>
+            }
+        >
+            <LemonTag type={tagType}>{display}</LemonTag>
```

Comment:
It would be so great if these could update live as well... I already had to manually refresh a few times.

Reply:
Yep this is part of the follow up PR. I just didn't want to continue bloating this one